### PR TITLE
added wrapper api for framework

### PR DESCRIPTION
**Pull request by anilkumarkammalapalli**

### CODE REVIEW

**Review by sharathkashyap — APPROVED** [review]

File: src/main/java/com/igot/cb/org/service/impl/OrgServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+
+    private @Autowired AccessTokenValidator accessTokenValidator;
+
+    private @Autowired DemandService demandService;
+
+    @Override
+    public ApiResponse readFramework(String frameworkName, String orgId, String userAuthToken) {
+        ApiResponse response = ProjectUtil.createDefaultResponse(Constants.API_ORG_CREATE);
+        try {
+            if (frameworkName == null || orgId == null) {
```
Comment (Sahil-tarento) [inline src/main/java/com/igot/cb/org/service/impl/OrgServiceImpl.java]: need to use StringUtils

File: src/main/java/com/igot/cb/org/service/impl/OrgServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+                return response;
+            }
+            String userId = accessTokenValidator.verifyUserToken(userAuthToken);
+            if (StringUtils.isBlank(userId)) {
+                response.getParams().setStatus(Constants.FAILED);
+                response.getParams().setErrMsg(Constants.USER_ID_DOESNT_EXIST);
+                response.setResponseCode(HttpStatus.BAD_REQUEST);
+                return response;
+            }
+            if (!demandService.isSpvRequest(userId,Constants.MDO_ADMIN)) {
```
Comment (Sahil-tarento) [inline src/main/java/com/igot/cb/org/service/impl/OrgServiceImpl.java:61]: Please do code formatting

File: src/main/java/com/igot/cb/org/service/impl/OrgServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+                response.getParams().setStatus(Constants.FAILED);
+                response.getParams().setErrMsg("User does not have the required role:");
+                response.setResponseCode(HttpStatus.BAD_REQUEST);
+                return response;
+            }
+            Map<String, Object> propertyMap = new HashMap<>();
+            propertyMap.put(Constants.ID, orgId);
+            List<Map<String, Object>> orgDetails = cassandraOperation.getRecordsByPropertiesWithoutFiltering(Constants.KEYSPACE_SUNBIRD, Constants.ORG_TABLE, propertyMap, null, 1);
+            if (orgDetails.isEmpty()) {
```
Comment (Sahil-tarento) [inline src/main/java/com/igot/cb/org/service/impl/OrgServiceImpl.java]: Use CollectionUtils

File: src/main/java/com/igot/cb/org/service/impl/OrgServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+                response.getParams().setStatus(Constants.FAILED);
+                response.getParams().setErrMsg("Organization not found");
+                response.setResponseCode(HttpStatus.BAD_REQUEST);
+                return response;
+            }
+            String fwName = (String) orgDetails.get(0).get(Constants.FRAMEWORKID);
+            if (fwName == null) {
```
Comment (Sahil-tarento) [inline src/main/java/com/igot/cb/org/service/impl/OrgServiceImpl.java]: Use StringUtils

File: src/main/java/com/igot/cb/org/service/impl/OrgServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+                String responseCode = (String) frameworkResponse.get(Constants.RESPONSE_CODE);
+                if (responseCode.equals(Constants.OK)) {
+                    Map<String, Object> result = (Map<String, Object>) frameworkResponse.get(Constants.RESULT);
+                    fwName = (String) result.getOrDefault(Constants.NODE_ID, "");
+                    Map<String, Object> map = new HashMap<>();
+                    map.put(Constants.FRAMEWORKID, fwName);
+                    map.put(Constants.ID, orgId);
+                    Map<String, Object> updateOrgDetails = cassandraOperation.updateRecord(Constants.KEYSPACE_SUNBIRD, Constants.ORG_TABLE, map);
+                    String updateResponse = (String) updateOrgDetails.get(Constants.RESPONSE);
+                    if (updateResponse.equalsIgnoreCase(Constants.SUCCESS)) {
```
Comment (Sahil-tarento) [inline src/main/java/com/igot/cb/org/service/impl/OrgServiceImpl.java]: Add null Pointer Exception

File: src/main/java/com/igot/cb/org/service/impl/OrgServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,183 @@
...
+            log.error("Failed to read framework with Id: " + frameworkId, e);
+            response.getParams().setErr("Failed to read framework: " + e.getMessage());
+            response.getParams().setStatus(Constants.FAILED);
+            response.setResponseCode(HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+        return response;
+    }
+
+
+
```
Comment (Sahil-tarento) [inline src/main/java/com/igot/cb/org/service/impl/OrgServiceImpl.java]: Remove this space

File: src/main/java/com/igot/cb/org/OrgController.java [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+@RequestMapping("/org")
+@Slf4j
+public class OrgController {
+
+    private @Autowired OrgService orgService;
+
+    @GetMapping(value = "/framework/read")
```
Comment (Sahil-tarento) [inline src/main/java/com/igot/cb/org/OrgController.java]: Remove value not required

File: src/main/java/com/igot/cb/org/service/impl/OrgServiceImpl.java [hunk 8]
```diff
@@ -0,0 +1,183 @@
...
+import java.util.*;
+
+@Service
+@Slf4j
+public class OrgServiceImpl implements OrgService {
+
+    @Autowired
+    private PayloadValidation payloadValidation;
+
+    private @Autowired CassandraOperation cassandraOperation;
```
Comment (Sahil-tarento) [inline src/main/java/com/igot/cb/org/service/impl/OrgServiceImpl.java]: Do proper code formatting and also private 
@autowired is wrong

File: src/main/java/com/igot/cb/org/service/impl/OrgServiceImpl.java [hunk 9]
```diff
@@ -0,0 +1,183 @@
...
+
+    private @Autowired OutboundRequestHandlerServiceImpl outboundRequestHandlerServiceImpl;
+
+    private @Autowired CbServerProperties cbServerProperties;
```
Comment (Sahil-tarento) [inline src/main/java/com/igot/cb/org/service/impl/OrgServiceImpl.java]: Remove private not required

File: src/main/java/com/igot/cb/org/service/impl/OrgServiceImpl.java [hunk 10]
```diff
@@ -0,0 +1,183 @@
...
+                response.getParams().setStatus(Constants.FAILED);
+                response.getParams().setErrMsg("Organization not found");
+                response.setResponseCode(HttpStatus.BAD_REQUEST);
+                return response;
+            }
+            String fwName = (String) orgDetails.get(0).get(Constants.FRAMEWORKID);
+            if (fwName == null) {
+                StringBuilder strUrl = new StringBuilder(cbServerProperties.getKnowledgeMS());
+                strUrl.append(cbServerProperties.getOdcsFrameworkCreate());
+                Map<String, Object> createReq = createFrameworkRequest(orgId, frameworkName);
```
Comment (Sahil-tarento) [inline src/main/java/com/igot/cb/org/service/impl/OrgServiceImpl.java:80]: Is this means you need to create a framwork in read request only

**Review by Sahil-tarento — CHANGES_REQUESTED** [review]

File: src/main/java/com/igot/cb/org/service/impl/OrgServiceImpl.java (hunk 10 above)
Comment (anilkumarkammalapalli) [reply src/main/java/com/igot/cb/org/service/impl/OrgServiceImpl.java:80]: yes if it is not there we need to create.

**Pull request merged**